include/kokkos-utils/meta/type_traits.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef KOKKOS_UTILS_META_TYPE_TRAITS_HPP
+#define KOKKOS_UTILS_META_TYPE_TRAITS_HPP
+
+#include <type_traits>
+
+/**
+ * @brief This namespace provides extensions to the @c type_traits standard header.
+ *
+ * References:
+ *      * https://en.cppreference.com/w/cpp/header/type_traits
+ */
+namespace Kokkos::utils::meta
```

Comment:
It could also become `Kokkos::utils::Impl` to collect more of such things. I.e. things that are not directly "kokkos utils". It could become a `impl` directory instead of `meta`. A bit like `Kokkos_Utilities.hpp` in `Kokkos/core/src/impl`.